src/frontend/src/binder/query.rs

```diff
@@ -362,18 +362,8 @@ impl Binder {
                 // bind the rest of the recursive cte
                 let mut recursive = self.bind_set_expr(*right)?;
 
-                // todo: add validate check here for *bound* `base` and `recursive`
                 Self::align_schema(&mut base, &mut recursive, SetOperator::Union)?;
-
-                // the final schema should be sticked with `recursive`'s
-                // e.g., <value> union all <select stmt>
-                let schema = recursive.schema().clone();
-                // yet another sanity check
-                assert_eq!(
-                    schema,
-                    recursive.schema().clone(),
```

Comment:
The original assertion always passes and looks like a typo, but `assert_eq` between `base` and `recursive` can actually fail because of name mismatch.
* The `align_schema` call above already ensures data types match[1].
* When names mismatch, PostgreSQL prefers the name from base query, so do we.

[1] Actually we need a special version of `align_schema`. I will post the detail in a new issue.